runtime/interpreter/value.go

```diff
@@ -7090,11 +7177,13 @@ func (v *DictionaryValue) IsModified() bool {
 		return true
 	}
 
-	if v.Keys.IsModified() {
+	v.ensureLoaded()
+
+	if v.keys.IsModified() {
 		return true
 	}
 
-	for pair := v.Entries.Oldest(); pair != nil; pair = pair.Next() {
+	for pair := v.entries.Oldest(); pair != nil; pair = pair.Next() {
 		value := pair.Value
 		if value.IsModified() {
 			return true
```

Comment:
Because of deferred decoding, is it possible for `v.keys.IsModified()` or `value.IsModified()` return true without DictionaryValue being decoded?

Reply:
Right, returning `isModified` as false, if the value is not loaded at all, makes sense.
```go
if v.content != nil {
    return false
}
```
We could do the same for composites and arrays as well.

Good suggestion! :1st_place_medal: